From 01652ec2961b10502b57824adb15478b6f112965 Mon Sep 17 00:00:00 2001
From: LeonardHerbst <leonard.herbst@tu-dresden.de>
Date: Fri, 26 Jun 2026 11:21:24 +0200
Subject: [PATCH] Added logging calls

---
 lparser.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lparser.c b/lparser.c
index 5894d8ff..5c24f859 100644
--- a/lparser.c
+++ b/lparser.c
@@ -27,6 +27,7 @@
 #include "lstring.h"
 #include "ltable.h"
 
+#include "lua_stlg.h"
 
 
 /* maximum number of local variables per function (must be smaller
@@ -1610,6 +1611,7 @@ static void statement (LexState *ls) {
 ** upvalue named LUA_ENV
 */
 static void mainfunc (LexState *ls, FuncState *fs) {
+  LOG_STAGE(parsecompile_start);
   BlockCnt bl;
   expdesc v;
   open_func(ls, fs, &bl);
@@ -1620,6 +1622,7 @@ static void mainfunc (LexState *ls, FuncState *fs) {
   statlist(ls);  /* parse main body */
   check(ls, TK_EOS);
   close_func(ls);
+  LOG_STAGE(parsecompile_end);
 }
 
 
-- 
2.51.0

